### PR TITLE
Port Bridge to Typescript

### PR DESCRIPTION
**Pull request by Half-Shot**
Depends on the remaining TS PRs, so #209 and #210.

This PR ports the last file to Typescript and thus completes the Typescript project. After this, we can safely merge #199

### CODE REVIEW

**Comment by jaller94** [comment]
Waiting for you to fix the tests.

File: src/components/room-upgrade-handler.ts [hunk 1]
```diff
@@ -12,102 +12,119 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
             }
             throw Error("Failed to handle upgrade");
-        })
+        }
     }
 
-    onInvite(ev) {
-        if (!this._waitingForInvite.has(ev.room_id)) {
+    // eslint-disable-next-line camelcase
+    public async onInvite(ev: {room_id: string}) {
```
Comment (jaller94) [inline src/components/room-upgrade-handler.ts:84]: Why is this returning a boolean and what's the information this boolean carries?

Comment (Half-Shot) [reply src/components/room-upgrade-handler.ts:127]: Have improved the JSDoc for the whole class.

**Review by jaller94 — APPROVED** [review]
Looks to be ok.
I found it especially though to compare bridge.js and bridge.ts.

**Pull request merged**